### PR TITLE
Return promise error if submit result is an immediate failure

### PR DESCRIPTION
**Pull request by clark800**

### CODE REVIEW

File: src/api/transaction/submit.js [hunk 1]
```diff
@@ -5,6 +5,16 @@ const validate = utils.common.validate;
 const Request = utils.common.core.Request;
 const convertErrors = utils.common.convertErrors;
 
+function convertSubmitErrors(callback) {
+  return function(error, data) {
+    if (data.engineResult !== 'tesSUCCESS') {
```
Comment (lumberj) [inline src/api/transaction/submit.js]: This isn't right. See here: https://ripple.com/build/transactions/#result-categories

The immediate response only indicates an immediate failure if the result code is `tej` or `tem`. Anything else means the transaction _may_ appear in a ledger.

Comment (clark800) [inline src/api/transaction/submit.js]: What's an example of a code that is not tesSUCCESS that could be a successfully submitted transaction?

Comment (lumberj) [inline src/api/transaction/submit.js]: Anything other than `tej` or `tem`

**Comment by clark800** [comment]
@lumberj updated

**Comment by lumberj** [comment]
LGTM

**Pull request merged**